### DISCUSSION TITLE
Datadog automatic integration

### DISCUSSION THREAD

**Discussion by szibis** [vectordotdev/vector#10214] [General]
It would be great to have Datadog integration - via an agent or via vector direct into Datadog (I think Enterprise Vector option will have that I suppose), but how. about OSS ?? 

I think tighter integration with Datadog would help for example in understanding and automatic building experience over metrics generated by Vector.

**Comment by jszwedko** [comment]
Hi @szibis !

A few points:

* You can use an `internal_metrics` source wired up to a `datadog_metrics` to ingest metrics into Datadog and build your own dashboards and alerts
* We’re working on out-of-the-box dashboards that you’ll eventually be able to migrate towards if you choose. These will be driven by the metrics you get from `internal_metrics`
* We’re working on an entirely new, paid product within Datadog that will tightly integrate with Vector to offer richer insight into, and management of, your Vector instances

Hopefully this helps! Let me know if you have any questions.

**Reply by szibis** [reply]
All this is great but I have `internal_mtrics` and the problem here is to be sure that what we see here is actually the things we produce in Vector. 
We need to have proper calculations and scale of metrics in proper time windows. 
Right now I am trying to compare data from Vector events vs metrics from the app producing that logs and data from vector are different.

The app is producing JSON log into stdout and in Vector I see a different number of events - cant compare them.
I made also some additional log2metrics at some transformation parts to compare and here number of log lines is even different than events from Vector and logs from an app perspective. 

All this makes me confused and would like to not consume a lot of time to compare those - I would like to know that what see from Vector I baseline and that this is for sure the truth and start using it as starting point in searching wy else is different - now I don't know for sure who is wrong.
